src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowPingSpec.groovy

```diff
@@ -0,0 +1,120 @@
+package org.openkilda.functionaltests.spec.flows.haflows
+
+import static org.junit.jupiter.api.Assumptions.assumeTrue
+import static org.openkilda.functionaltests.extension.tags.Tag.LOW_PRIORITY
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.HaFlowHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.model.SwitchId
+import org.openkilda.model.cookie.Cookie
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPingPayload
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPingResult
+import org.openkilda.northbound.dto.v2.yflows.SubFlowPingPayload
+import org.openkilda.northbound.dto.v2.yflows.UniSubFlowPingPayload
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Narrative
+import spock.lang.Shared
+import spock.lang.Ignore
+
+@Narrative("""This spec tests 'periodic ping' functionality.""")
+class HaFlowPingSpec extends HealthCheckSpecification {
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Value('${flow.ping.interval}')
+    int pingInterval
+
+    @Tidy
+    @Ignore("https://github.com/telstra/open-kilda/issues/5217")
+    def "Able to turn on periodic pings on a ha-flow"() {
+        when: "Create a ha-flow with periodic pings turned on"
+        def swT = topologyHelper.switchTriplets.first()
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT).tap {
+            it.periodicPings = true
+        }
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+
+        then: "Periodic pings is really enabled"
+        northboundV2.getHaFlow(haFlow.haFlowId).periodicPings
+
+        and: "Packet counter on catch ping rules grows due to pings happening"
+        def sharedSwitchPacketCount = getPacketCountOfVlanPingRule(swT.shared.dpId)
```

Comment:
Please, move to `haFlowHelper.periodicPingPacketCountGrowsOnAllEndpoints()`

Reply:
If you leave this method in this class - it's obvious, that nobody else won't ever know it exists at all (which would really lead to code duplication when someone needs this method).